core/src/test/java/kafka/test/junit/LeakTestingExtension.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.test.junit;
+
+import kafka.utils.TestUtils;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+public class LeakTestingExtension implements AfterEachCallback {
+    @Override
+    public void afterEach(ExtensionContext extensionContext) {
+        TestUtils.verifyNoUnexpectedThreads("@AfterEach");
```

Comment:
Thanks @wernerdv . I had a basic question about this change - 
(TestUtils.verifyNoUnexpectedThreads)[https://github.com/apache/kafka/blob/trunk/core/src/test/scala/unit/kafka/utils/TestUtils.scala#L2432] seems to check for the presence of specific type of threads. Will this check be sufficient to catch all types of thread leaks ?